src/main/java/gateways/Tags/TagDatabaseGateway.java

```diff
@@ -0,0 +1,77 @@
+package gateways.Tags;
+
+import entities.Tag;
+import gateways.DatabaseGateway;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class TagDatabaseGateway extends DatabaseGateway implements TagDataAccessInterface {
+
+    private ArrayList<Tag> tagList;
+
+    /**
+     * Instantiating DatabaseGateway with a relative path will store the absolute path to the respective database.
+     * If the file at the path does not exist, one will be created
+     * This absolute path can be fetched using the getAbsoluteFilepath method
+     *
+     * @param relativePath The relative path to the database
+     * @throws IOException Throws exception if encountering failed or interrupted IO exceptions
+     */
+    public TagDatabaseGateway(String relativePath) throws IOException, ClassNotFoundException {
+        super(relativePath);
+        load();
+    }
+
+    @Override
+    public Tag get(int id) {
```

Comment:
Autograding threw errors for all @Override methods except for getAll, because it did not "implement or override a corresponding supertype method". I think for inser, delete and edit these might be caused by the autograder bugging out due to the existing errors, but it is true that TagDataAccessInterface does not have a get method, so the Override is unnecessary (as of this version).